File: 2025-06-June-GFG-POTD/June 14 - Symmetric Tree/Solution.py
class Node:
    def __init__(self, data):
        self.data = data
        self.left = None
        self.right = None

class Solution:
    def isSymmetric(self, root):
        if not root:
            return True
        return self.isMirror(root.left, root.right)

    def isMirror(self, t1, t2):
        if not t1 and not t2:
            return True
        if not t1 or not t2:
            return False
        if t1.data != t2.data:
            return False
        return self.isMirror(t1.left, t2.right) and self.isMirror(t1.right, t2.left)

File: 2025-06-June-GFG-POTD/June 14 - Symmetric Tree/Solution.cpp
/*
struct Node {
    int data;
    Node* left;
    Node* right;
    Node(int val) {
        data = val;
        left = nullptr;
        right = nullptr;
    }
};
*/

class Solution {
public:
    bool isSymmetric(Node* root) {
        if (!root) return true;
        return isMirror(root->left, root->right);
    }

private:
    bool isMirror(Node* t1, Node* t2) {
        if (!t1 && !t2) return true;
        if (!t1 || !t2) return false;
        if (t1->data != t2->data) return false;
        return isMirror(t1->left, t2->right) && isMirror(t1->right, t2->left);
    }
};
